Integrate messaging page / bubbles

lhlee-umass opened the pull request:
- Develop message bubble as reusable component
- Display new message bubbles on MessageSellerPage
- Link Home to MessageSellerPage when Message Seller button on a listing card is clicked
- Added additional routing path "/message-seller"

This PR will result in finishing the following issues: 
Closes https://github.com/lhlee-umass/streetside/issues/6 (main issue)
Closes https://github.com/lhlee-umass/streetside/issues/9 (sub-issue)
Closes https://github.com/lhlee-umass/streetside/issues/10 (sub-issue)

lauriehlee committed:
Integrate messaging page / bubbles

- Develop message bubble as reusable component
- Display new message bubbles on MessageSellerPage
- Link Home to MessageSellerPage when Message Seller button on a listing card is clicked
- Added additional routing path "/message-seller"

zhandavidz commented on frontend/src/components/listing-card/ListingCard.tsx:
```diff
@@ -1,5 +1,6 @@
 import React from 'react'
 import { Heart, MapPin, Tag } from 'lucide-react'
+import { Link } from 'react-router-dom'
 
```
We're using the `react-router` package rather than the `react-router-dom` package, so the import should use `react-router` instead (I believe `react-router-dom` used to be widely used but is now deprecated in favor of just `react-router`).

The functionality of `Link` remains identical, however, so only the import line should need to be changed.

lhlee-umass replied:
Great point! Changed it to 'react-router'

zhandavidz commented on frontend/src/App.tsx:
I don't think a change in this file is necessary, `main.tsx` already has `<Home />` as a index child of the root `/` path so with `<Outlet />` in `App.tsx`, we can render all pages into the `<Outlet />` component, and it should have the same functionality as having App just return `<Home />`.

What do you think? I think it may be better to leave `App.tsx` returning an `<Outlet />`, as it allows us to force certain commonalities on all pages, such as including a `<NavBar />` component before the `<Outlet />` to stick it above all pages.

lhlee-umass replied:
Got it, should be reverted

zhandavidz reviewed (changes requested):
Mostly LGTM. I left a few notes about `react-router-dom` vs `react-router` and about using `Outlet`'s in `App.tsx` that should be quick fixes.

Also, I think we probably should eventually have just a `/messages` page that allows the user to select from a conversations side bar and access specific conversations through a dynamic `/messages/:user_id` route, but for now the `/message-seller` is fine since we have none of the backend/api data and functionality to support actually mocking messaging yet.

Great work!

lhlee-umass committed:
Modified import statement and App.tsx

- Changed import from 'react-router-dom' to 'react-router' since its deprecated
- Better to leave App.tsx returning an Outlet, as David mentions it allows us to force certain commonalities on all pages

zhandavidz reviewed (approved):
The fixes look great! I think it's ready to merge!
